superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover.jsx

```diff
@@ -123,6 +123,9 @@ export default class AdhocMetricEditPopover extends React.PureComponent {
         adhocMetricLabel: this.state.adhocMetric?.getDefaultLabel(),
       });
     }
+    if (prevProps.datasource !== this.props.datasource) {
```

Comment:
why are we comparing the whole **datasource** object here instead of comparing the **datasource.name** only like here: https://github.com/apache/superset/pull/12782/files#diff-b6ef883f6b46721ba01ae63a0167874a10de1ce512b3fe4cc3535b9df6d19c2eR73 ?

Reply:
This does refer's to the the datasource name. datasource is datasource name being passed down